plugins/outputs/signalfx/signalfx.go

```diff
@@ -383,22 +219,43 @@ func (s *SignalFx) GetObjects(metrics []telegraf.Metric, dps chan *datapoint.Dat
 		var metricType datapoint.MetricType
 		var metricTypeString string
 
-		metricType, metricTypeString = parseMetricType(metric)
+		// patch memory metrics to be gauge
+		if metric.Type() == telegraf.Counter && metric.Name() == "mem" {
```

Comment:
Can't you just fix the mem plugin to emit the right type since we control this fork?

Reply:
No I am purposely not modifying any upstream plugins if at all possible.

Reply:
What is actually converting it to a counter?  The original metric seems to be a gauge: https://github.com/signalfx/telegraf/blob/signalfx-integration/plugins/inputs/system/memory.go#L40